source/Preferences.cpp

```diff
@@ -365,9 +371,15 @@ bool Preferences::ZoomViewIn()
 
 bool Preferences::ZoomViewOut()
 {
-	if(zoomIndex == 0)
+	const auto &zooms = GameData::Interfaces().Get("main view")->GetList("zooms");
+	if(!zoomIndex || zooms.size() <= 1)
 		return false;
 
+	// Make sure that we're actually zooming out. This can happen if the zoom index
+	// is out of range.
+	if(zoomIndex >= zooms.size())
+		zoomIndex = zooms.size() - 1;
+
 	--zoomIndex;
```

Comment:
Say that `zooms.size() == 1` and `zoomIndex == 4`. We snap zoomIndex to `zooms.size() - 1 = 0`, then do `--zoomIndex = -1`. But zoomIndex is a size_t now, which is an unsigned int, meaning this'll wrap to some large value.

I feel like making sure that zoomIndex is within the proper range should be done somewhere else, perhaps in a Preferences::FinishLoading function.

Reply:
The zoom list of size 0 or 1 is caught in an if before this happens.